### DISCUSSION TITLE
How to make Tsx file read the same types as Vue SFC does?

### DISCUSSION THREAD

**Discussion by mockingjet** [vuejs/language-tools#1873] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/32839503/190325117-093d563a-1419-4321-8c62-bcb4e1015fb6.png)

In Vue file, we can see the SfcButton props types.

![image](https://user-images.githubusercontent.com/32839503/190325210-55da36e6-1f79-4b07-a80c-388cac43490c.png)

But in Tsx file, we can only see the default SFC types.
```ts
// global.d.ts

declare module '*.vue' {
	import type { DefineComponent } from 'vue';
	const component: DefineComponent<{}, {}, any>;
	export default component;
}
```
How to make tsx read the correct prop types?

**Comment by johnsoncodehk** [comment, marked as answer]
See https://vuejs.org/guide/typescript/overview.html#volar-takeover-mode.

**Reply by mockingjet** [reply]
It works, thanks!
